src/cephadm/cephadmlib/constants.py

```diff
@@ -12,7 +12,7 @@
 DEFAULT_GRAFANA_IMAGE = 'quay.io/ceph/grafana:10.4.0'
 DEFAULT_HAPROXY_IMAGE = 'quay.io/ceph/haproxy:2.3'
 DEFAULT_KEEPALIVED_IMAGE = 'quay.io/ceph/keepalived:2.2.4'
-DEFAULT_NVMEOF_IMAGE = 'quay.io/ceph/nvmeof:1.2.1'
+DEFAULT_NVMEOF_IMAGE = 'quay.io/ceph/nvmeof:1.2'
```

Comment:
should we be too loose with the pinning? especially a critical component like nvmeof which is subject to change a lot since its still in its early days? And the version changing will affect other components like the dashboard and external clients which rely on the grpc apis that are associated with nvmeof.

Reply:
on a stable branch I'd be entirely against this. On main, I'm a bit more lenient, although the fact we've gone through 17 z releases for this already in not that much time does make me wonder if it's a bit too much churn even for main

Reply:
We do plan to merge specific versions to stable branches. For main, since we would have to make a PR for every new build so we thought we can keep it 1.2. I think all changes would be backwards compatible so hopefully shouldn't cause too much trouble on main. If you prefer we don't then we can switch to specific version on main too. 
cc: @caroav

Reply:
I understand the concerns raised here. We could change it to be a specific version. But that means that we need to remember to update it from time to time. 
Last year, the nvmeof significantly changed that's why we had so many versions. I'm not sure its going to be the same way in the future, but possibly yes. 
@adk3798 @ktdreyer @nizamial09 if you think it makes more sense to pin a specific version, we will make this change.

Reply:
my concern with the floating tag is if there is enough churn the teuthology tests for nvmeof could suddenly start failing without changes within the ceph codebase. If we update the image to a specific one we can verify with each new release we try whether it works. But I personally wouldn't block using the floating tag in main, especially if you think the rate of releases will slow down.